fix: prevent deinit callback from being called twice

hauleth opened the pull request:
This also fixes return value from `register_before_send/2` callback. By the API it should always return `Plug.Conn.t()` while we were returning `:ok`.

Ziinc commented:
@hauleth looks good, could you also bump the patch version ( in the readme as well)?

hauleth committed:
fix: prevent deinit callback from being called twice

This also fixes return value from `register_before_send/2` callback. By
the API it should always return `Plug.Conn.t()` while we were returning
`:ok`.
